### DISCUSSION TITLE
Problema Integrazione in home assistant di zbbridge e mqtt

### DISCUSSION THREAD

**Discussion by n3dv3d** [indomus/forum#996] [HUB: Home Assistant]
Salve a tutti,

ho installato tasmota sullo zbbridge e configurato mqtt con broker installato su raspberry, ho effettuato il pairng bridge-dispositivi, vedo che il broker e il bridge comunicano correttamente ( mettendomi in ascolto dal plugin mqtt di home assitsant ricevo i dati dal bridge), ma non riesco a creare le entità da utilizzare in home assistant... avete qualche dritta da darmi su come dovrei fare per integrare i dispositivi associati al bridge in home assistant? Ho provato a settare così il file di configurazione, ma non funziona:
```
mqtt:
  broker: "127.0.0.1"
  port: 1883
  username: "mqtt_user"
  password: `"pwd"`


sensor: 

- platform: mqtt 
  name: "temp_sensor" 
  state_topic: "tele/sonoff-z2t/SENSOR/"
  value_template: "{{ value_json.ZbReceived['0x4923'].Temperature }}" 
  unit_of_measurement: "°C" 
  device_class: temperature
```

Avete suggerimenti?
Grazie

**Comment by snakuzzo** [comment]
[QUI](https://indomus.it/progetti/realizzare-un-bridge-gateway-zigbee-mqtt-autonomo-con-nodemcu-e-cc2530-e-affini/) avevamo postato qualche esempio, ma quella guida è vecchiotta e comunque sono state fatte parecchie modifiche lato Tasmota.
Sul sito ufficiale trovi un'ottima panoramica su come funziona la parte Zigbee (non solo per i sensori, ma anche per pilotare altri device come ad esempio switch, light, etc) -> [LINK](https://tasmota.github.io/docs/Zigbee/).
Sempre sul sito ufficiale trovi anche i comandi relativi alla parte Zigbee -> [LINK](https://tasmota.github.io/docs/Commands/#zigbee)
Attento che sempre lì ci sono vari setOption che ti permettono di seplificare la struttura dei topic (rendendoli univoci...cosa che di default non sono)...stesso dicasi per i payload

Resta il fatto che per poter effettuare queste configurazioni a mano devi conoscere perfettamente come funziona il protocollo MQTT e la parte di template su HA per capire come "parsare" il Payload

**Reply by snakuzzo** [reply]
Se non sei particolarmente pratico e vuoi continuare ad usare il ZB Bridge, ti consiglierei di utilizzarlo con ZHA

**Reply by essegi65** [reply]
io da circa 1 mese ho installato e flashato un sonoff bridge zigbee e integrato a HA con ZHA (come detto da @snakuzzo) per adesso in prova ho collegato 4 device (2 prese tuya zigbee, 1 sensore aqara e un comando presa bticino living now) il tutto funziona e per adesso non ho problemi nemmeno di connessione. 2 device sono in soggiorno con il GW e 2 in cucina (separati da un vano scala con porta)

**Comment by n3dv3d** [comment]
Buongiorno e grazie a tutti per le risposte, in realtà lo uso già con ZHA e funziona bene, ora volevo collegarci 2 valvole termostatiche (smart life) che mi hanno regalato, ma purtroppo anche se rilevate correttamente non mi dà entità da utilizzare; così avevo pensato di provare un approccio diveso, con mqtt... pensando che visto che bridge e broker comunicavano correttamente fosse più semplice integrare tutto in home assistant, ma dalle vostre risposte ho idea di essermi sbagliato :(
Ho anche provato ad utilizzare i plugin tuya di home assistant, sia local che via cloud ma, anche quelli. non riesco a farli funzionare, a causa di errori vari di autenticazione ( invalid auth nel primo caso e skill id invalid nel secondo) leggendo su google tanti sembrano avere questi errori dovuti alla piattaforma iot di tuya e non ho trovato modo di risolvere.

Avete qualche altro suggerimento per integrare le valvole in home assistant? come mai ZHA me le rileva, ma senza entità da utilizzre?

Grazie a tutti e buona giornata

**Reply by snakuzzo** [reply]
per la domanda su ZHA dovresti aprire una issue sul progetto homeassistant core su github...noi non siamo in grado di dirti il perchè.
Ti consiglierei di passare a Zigbee2mqtt, ma con il ZB Bridge mi sa che non puoi. Ti tocca metterti lì e seguire le indicazioni di Tasmota ce ti ho indicato, altrimenti dovresti cambiare coordinator.

P.S. rispondi sotto l'ultimo messaggio, non creare nuovi thread come hai fatto ora. Grazie

**Reply by n3dv3d** [reply]
Ok, scusa mi ero accorto dopo aver risposto :)
Comunque avevo provato anche ad installare zigbee2mqtt sul raspberry dove ho home assistant core installato, ma mi dà errore durante l'installazione... 
A questo punto mi sorge spontanea una domanda, sarebbe possibile utilizzare zigbee2mqtt anche con home assistant core oppure bisogna avere la versione supervised? 

Io avrei a disposizione anche un bridge smartlife, credi che sarebbe più facile integrarlo in home assistant rispetto al ZB Bridge?

Grazie

**Reply by snakuzzo** [reply]
Sono due cose completamente scollegate. Se hai avuto errore durante l'installazione non vuol dire che non funzioni con Core :D
Le cose non si risolvono così... "mi dà errore allora cambio strada", ma si cerca l'errore e si tenta di risolvere. 

Quanto all'hub smartlife, non credo (ma non sono sicuro) si possa utilizzare su HA

**Reply by n3dv3d** [reply]
ok, grazie mille.. cercherò di concentrarmi su zigbee2mqtt e vediamo cosa riesco a fare.

**Reply by snakuzzo** [reply]
Comunque...se posso permettermi...se non sei abbastanza pratico -> HassOS
In caso diverso passa ad installazione Core su Docker

**Reply by manny199** [reply]
Ciao a tutti, uso zbbridge con Tasmota da qualche giorno con l'integrazione di HA Core **Zigbee** Home **Automation.**
Purtroppo il bridge è molto instabile e spesso si disconnette manifestando numerosi errori nel log.
Esiste qualche altra soluzione che superi questi problemi di instabilità?

**Reply by essegi65** [reply]
Personalmente sto usando la tua stessa configurazione da circa 3/4 settimane con 5 device collegati (prese tipo schuko, sensori pir, prese bticino) e francamente non ho mai avuto instabilità di connessione. 2 prese tanto per dire sono in un'altra stanza rispetto al bridge e il segnale arriva, un sensore pir è in una veranda esterna e anche li nessun problema di segnale. Sicuro non ci siano interferenze con il tuo wifi?

**Reply by snakuzzo** [reply]
@manny199 ma poi di che errori parliamo ?
E' un problema a livello zigbee o a livello di coordinator ?
Dovresti provare ad indagare prima di cercare soluzioni alternative

**Reply by manny199** [reply]
@snakuzzo Ciao, gli errori sono:
```
(bellows.thread_0) [bellows.uart] CRC error in frame b'38e4a1a9a52a70b23f946d25b85597499c4e27dbd5d269381c0e07bbe5ca6e' (b'0b99' != b'd56b')
2021-11-11 06:18:41 ERROR (MainThread) [bellows.ezsp] NCP entered failed state. Requesting APP controller restart
Logger: bellows.zigbee.application Source: /srv/homeassistant/lib/python3.9/site-packages/bellows/zigbee/application.py:709 First occurred: 09:39:08 (1 occurrences) Last logged: 11:46:08  Watchdog heartbeat timeout:
```
Firmware Tasmota 10.0.0
File OTA per zigbbrdge: ncp-uart-sw-6.7.8_115200.ota

@essegi65 
Il router / access point è ad un metro dal bridge, per cui il segnale deve essere super; l'ho anche fatto agganciare ad un altro access point ma continuano a persistere gli errori sopra evidenziati. Peraltro ci sono vari articoli che parlano di forti instabilità del sonoff zbbridge, vedi:
https://www.home-assistant.io/integrations/zha/#warning-about-wi-fi-based-zigbee-to-serial-bridgesgateways
https://github.com/home-assistant/home-assistant.io/issues/17170

Grazie a tutti

**Reply by n3dv3d** [reply]
> Comunque...se posso permettermi...se non sei abbastanza pratico -> HassOS In caso diverso passa ad installazione Core su Docker

Grazie mille per il tuo suggerimento, installato HassOS e zigbee2mqtt e sono riuscito a far funzionare tutte le valvole tramite zbbridge, ora però ho un piccolo problema... purtroppo spesso, e non riesco a capire il motivo, quando raggiungono la temperatura target non si stoppano e la valvola continua a rimanere aperta, così ho pensato di fare un'automazione di controllo per fermarla se supera una certa temperatura, questa è l'automazione:


```
alias: 'Bagno ' 
description: 
trigger:
  - platform: numeric_state
    entity_id: sensor.sensore_temperature
    attribute: temperature
    above: '20'
condition: []
action:
  - service: script.close_bagno
mode: restart

```

funziona abbastanza bene, ma volevo sostituire il 20 del campo above con l'attributo current_heating_setpoint così da non avere un valore fisso, ma fargli usare il valore target che imposto.. ovviamente ho provato a modificarlo inserndo l'attributo, ma non funziona... avete qualche suggerimento? 

Grazie mille

**Reply by snakuzzo** [reply]
Innanzi tutto correggi il post usando il markdown per il codice.
Poi...io credo che il comportamento del termostato sia normale (attenzione, parlo del termostato e non di HA). Solitamente non si chiudono subito, ma dopo un po'.

Quanto all'automazione...non puoi limitarti a dire "non funziona".
Devi fare delle prove e vedere dai log cosa c'è che non va...c'è anche la funzionalità di debug da GUI se imposti l'id dell'automazione

Innanzi tutto devi provare a vedere se scatta il trigger...e per farlo, in base a quello che hai scritto, dovresti forzare il valore dell'attributo a passare da un valore inferiore a 20 ad uno superiore a 20. Se è già sopra 20 non scatterà mai.

**Reply by n3dv3d** [reply]
Ok, post corretto, sorry.
Comunque l'automazione postata sopra funziona correttamente, volevo solo sapere se fosse possibile nel campo above inserire un attributo invece di un valore numerico, perchè se provo a salvare inserendo l'attributo current_heating_setpoint nel campo above mi dà subito errore di sintassi e non mi fa neanche salvare per lanciare l'automazione... 
Per quanto riguarda il termostato non credo sia normale,  è capitato anche che salisse di più di 3 gradi la temperatura senza fermarsi, e me lo fa su 2 valvole su 2 quindi presumo possa essere qualche problema nel firmware?!
Grazie

**Reply by snakuzzo** [reply]
[QUI](https://www.home-assistant.io/docs/automation/trigger/#numeric-state-trigger) trovi qualche esempio...ma il punto è "che errore ti dà ?" Quando dite che una cosa non funziona o dovete dare evidenza di quanto accade. Che logga ? Che errore ti dà ? Per favore siate più "precisi" e dettagliate il più possibile.
Sul problema che non chiude...l'automazione su HA è solo una pezza perché, come hai detto tu in merito al firmware, è proprio la testa termostatica che funziona male

**Reply by n3dv3d** [reply]
Se provo a salvare mettendo l'attributo current_heating_setpoint nel campo above mi dà questo errore quando clicco su salva: 
Message malformed:expected float for dictionary value @data ['above'] e nel log non c'è nulla.
Ho già visto quel post, ma mi sembra che il campo above sia solo numerico, non è possibile passargli un attributo? c'è qualche altro modo per lanciare uno script quando sensor.sensore_temperature > current_heating_setpoint ?

Sono d'accordo che sia solo una pezza, ma credo che impostando bene l'automazione siano comunque utilizzabili..diciamo che piuttosto che lasciarle nell' armadio preferisco usarle con una pezza :)

Grazie

**Reply by snakuzzo** [reply]
Per `above` si aspetta un float...mentre tu gli stai passando qualcos'altro ... ovvero una `'stringa'`.
Confronta quello che hai scritto con quello che c'è nell'esempio... :)

**Reply by n3dv3d** [reply]
esatto, non capisco come fargli fare un confronto tra due attributi e fargli eseguire l'azione solo nel caso uno sia maggiore dell'altro... non mi sembra di aver visto esempi simili a quello che vorrei fare..hai qualche dritta da darmi?

**Reply by snakuzzo** [reply]
Però i trigger non è che li fai "ad occhio" :)
Il parametro `above` del trigger `numeric_state` si aspetta un float.
Nel tuo caso dovresti usare un [template trigger](https://www.home-assistant.io/docs/automation/trigger/#template-trigger) e... seguendo quanto scritto in documentazione...

> Template triggers work by evaluating a template when any of the recognized entities change state. The trigger will fire if the state change caused the template to render ‘true’ (a non-zero number or any of the strings true, yes, on, enable) when it was previously ‘false’ (anything else).

...puoi provare a fare così...

```yaml
trigger:
  - platform: template
    value_template: "{{ states('sensor.sensore_temperature') | float > state_attr('sensor.sensore_temperature','current_heating_setpoint') | float }}"
    for:
      minutes: 2
```

Quindi... nel momento in cui il template restituirà `true` (e puoi testarlo direttamente su Lovelace sotto `Developer tools > Template`), scatterà il trigger (da notare il for di 2 minuti perché se la temperatura dovesse subito passare sotto soglia e poi subito sopra soglia...scatterebbe due volte)
Inoltre ho usato il filter `float` perché perché devi fare il confronto tra due valori numerici, mentre l'output dei template è sempre una stringa.

Dai anche un'occhiata ai `default values` quando usi questi filter -> [QUI](https://www.home-assistant.io/docs/configuration/templating/#important-template-rules)

**Reply by n3dv3d** [reply]
Grazie mille, gentilissimo.. nel week end provo e ti faccio sapere.
